pkg/rc/farm.go

```diff
@@ -65,9 +66,9 @@ func NewFarm(
 // Start is not safe for concurrent execution. Do not execute multiple
 // concurrent instances of Start.
 func (rcf *Farm) Start(quit <-chan struct{}) {
-	consulutil.WithSession(quit, rcf.sessions, func(sessionQuit <-chan struct{}, session string) {
-		rcf.logger.WithField("session", session).Infoln("Acquired new session")
-		rcf.lock = rcf.kpStore.NewUnmanagedLock(session, "")
+	consulutil.WithSession(quit, rcf.sessions, func(sessionQuit <-chan struct{}, sessionID string) {
+		rcf.logger.WithField("session", sessionID).Infoln("Acquired new session")
+		rcf.session = rcf.kpStore.NewUnmanagedSession(sessionID, "")
```

Comment:
The terminology gets a little confusing here... `WithSession()` is creating the underlying consul session, we just want a `kp.Session` that uses it. If anyone has suggestions to increase clarity i'm all ears